qml/FileListPage.qml

```diff
@@ -43,48 +57,99 @@ Page {
+                //% "Documents"
+                title: qsTrId("sailfish-office-he-apptitle")
+            }
+
+            SearchField {
+                id: searchField
 
-            // TODO: uncomment once there are more document sources
-            // title: page.title
+                width: parent.width
+                opacity: page.searchEnabled ? 1.0 : 0.0
+                visible: opacity > 0
 
-            Binding {
-                target: page
-                property: "searchText"
-                value: header.searchText
+                //: Document search field placeholder text
+                //% "Search documents"
+                placeholderText: qsTrId("sailfish-office-tf-search-documents")
+
+                // We prefer lowercase
+                inputMethodHints: Qt.ImhNoAutoUppercase | Qt.ImhPreferLowercase | Qt.ImhNoPredictiveText
+                EnterKey.iconSource: "image://theme/icon-m-enter-close"
+                EnterKey.onClicked: focus = false
+
+                Behavior on opacity { FadeAnimation { duration: 150 } }
+                Behavior on height {
+                    NumberAnimation {
+                        duration: 150
+                        easing.type: Easing.InOutQuad
+                    }
+                }
+                Binding {
+                    target: searchField
+                    when: !searchEnabled
+                    property: "height"
+                    value: 0
+                }
             }
+        }
 
-            Connections {
-                target: menuItemSearch
-                onClicked: header.enableSearch()
+        Connections {
+            target: searchField.activeFocus ? listView : null
+            ignoreUnknownSignals: true
+            onContentYChanged: {
+                if (listView.contentY > (Screen.height / 2)) {
+                    searchField.focus = false
+                }
             }
         }
 
         PullDownMenu {
-            visible: listView.count > 0
-            MenuItem {
-                id: menuItemSearch
+            id: menu
+
+            property bool _searchEnabled
 
-                //: Search menu entry
-                //% "Search"
-                text: qsTrId("sailfish-office-me-search")
+            // avoid changing text state while menu is open
+            onActiveChanged: {
+                if (active) {
+                    _searchEnabled = page.searchEnabled
+                }
+            }
+
+            MenuItem {
+                text: !menu._searchEnabled ? //% "Show search"
+                                             qsTrId("sailfish-office-me-show_search")
+                                             //% "Hide search"
+                                           : qsTrId("sailfish-office-me-hide_search")
+                onClicked: page.searchEnabled = !page.searchEnabled
             }
         }
-        
-        ViewPlaceholder {
+
+        InfoLabel {
```

Comment:
ViewPlaceHolder has a click animation which doesn't make sense here. Pulldown menu doesn't include any item that allows to create more content. Plus on search case even that wouldn't make sense.

Custom y value is another thing. No existing documents could have the label in the middle, but for search case it's better to have search indication closer to the search field.